### ISSUE TITLE
Option to add notes at specific points in time

### ISSUE THREAD

**Issue by evmaki** [grimmdude/MidiWriterJS#39]
First off: fantastic work on this! Thank you for all the time you've put in, and for making it available for everyone to use. I've used it in many projects already. 

One thing that has tripped me up a few times is the inability to add notes at specific points in time. Say, for instance, I already have all of my notes in some data structure somewhere and I already know where, exactly, I want all of them to go. I've always had to do awkward, hacky stuff with the `wait` and `duration` parameters in order to do this.

That being said, two questions:
1. Is there an easier way of doing this that I'm missing?
2. If not, would you consider this a useful feature? I'd like to add it to the library and get a pull request going if so.

**Comment by grimmdude** [comment]
Hey @evmaki,

When you say you would like to add notes at a specific "time" are you referring to minutes/seconds?  The library doesn't support that exactly, but there is a duration value you can use which places the note a specific tick.

With a fairly trivial calculation you should be able to translate that into a time.  You would just prefix the `duration` value with a `T`.  Would look something like `{duration: 'T4550'}`.

**Example:**
MidiWriterJS defaults to 128 ticks per beat.  If your song is 120bpm there are 15360 ticks every minute (tpb * bpm), so if you want to place a note at the 90 second mark you would do something like this:
```
tpb * bpm / 60 * seconds = tick number
128 * 120 / 60 * 90 = 23040

// {duration: 'T23040'} would place the note at 1:30 in 120bpm
```
Let me know if you have any questions.

-Garrett

**Comment by evmaki** [comment]
@grimmdude thanks for your quick response.

By time I mean the number of ticks into the track you want the note to start at, which would translate easily into arrangement position (or however you refer to the way bars and beats are counted in your DAW of choice)

Working with your above example, I've got:
```
var track = new MidiWriter.Track()
track.addEvent(new MidiWriter.NoteEvent({pitch: 'C5', duration: 'T4550'}))
// do stuff to save to disk...
```

This gives me a note that sounds for 4550 ticks, but not one that starts 4550 ticks into the track (with 4550 ticks of silence before it). I _can_ accomplish this by doing:

```
track.addEvent(new MidiWriter.NoteEvent({pitch: 'C5', duration: '4', wait: 'T4550'})
```

But when you're trying to build a track from some other data and the notes aren't legato, there's some boilerplate involved in being able to place them in their correct positions using a mixture of the correct `duration` and `wait` values.

Ideally, I could have a whole bunch of notes with specified start times (in ticks) and, using `addEvent` and some parameter, end up with a midi file with notes that have the correct rests between them. The "Hot Cross Buns" example in the documentation might look like this:
```
track.addEvent([
			new MidiWriter.NoteEvent({pitch: ['E4','D4'], start: 'T0', duration: '4'}),
			new MidiWriter.NoteEvent({pitch: ['C4'], start: 'T512', duration: '2'}),
			new MidiWriter.NoteEvent({pitch: ['E4','D4'], start: 'T768', duration: '4'}),
			new MidiWriter.NoteEvent({pitch: ['C4'], start: 'T1280', duration: '2'}),
			new MidiWriter.NoteEvent({pitch: ['C4', 'C4', 'C4', 'C4', 'D4', 'D4', 'D4', 'D4'], start: 'T1536', duration: '8'}),
			new MidiWriter.NoteEvent({pitch: ['E4','D4'], start: 'T2560', duration: '4'}),
			new MidiWriter.NoteEvent({pitch: ['C4'], start: 'T3072', duration: '2'})
	], function(event, index) {
    return {sequential:true};
  }
);
```
Here I've left the notes legato, so in this context it's unnecessary–but since I've specified the start location, they _could_ be staccato or something else without having to go through the work of calculating rest times.

This would, however, introduce an issue if you attempt to add notes out of order.

**Comment by grimmdude** [comment]
Hi @evmaki,

Ah, right I don't know what I was thinking; duration is not the right parameter for this.  I see your point, you need a `start` option in which the tick value is unrelated to the previous note (unlike `wait`).  I will take a look at this a little later and see what I can come up with.  Thanks,

-Garrett

**Comment by muggy8** [comment]
Hi there, great work! I just wanna say that I would really appreciate this feature too!

**Comment by benrogmans** [comment]
Yep, me too! 😁

**Comment by grimmdude** [comment]
@muggy8, @benrogmans,

Appreciate the feedback and interest, I will work on this as soon as I can.

-Garrett

**Comment by marcohamersma** [comment]
Sorry to pile up on this, but I'm also looking forward to this feature. The thing I'm trying to make has the occasional notes from different channels playing at the same time, so being able to have an API that allows you to create a note at a certain time would be very welcome.

**Comment by grimmdude** [comment]
Hi @evmaki, @muggy8, @benrogmans, @marcohamersma,

This feature is now available in [1.7.0](https://github.com/grimmdude/MidiWriterJS/releases/tag/1.7.0)!

Give it a shot and please let me know your results.

-Garrett
